lib/kennel/models/record.rb

```diff
@@ -119,6 +123,21 @@ def remove_tracking_id
         self.class.remove_tracking_id(as_json)
       end
 
+      def build_json
+        {
+          id: id
+        }.compact
+      end
+
+      def as_json
+        @as_json ||= begin
+                       json = build_json
+                       (id = json.delete(:id)) && json[:id] = id
+                       validate_json(json) if validate
+                       json
+                     end
```

Comment:
consistent indent is nice :)
```suggestion
        @as_json ||=
          begin
            json = build_json
            (id = json.delete(:id)) && json[:id] = id
            validate_json(json) if validate
            json
          end
```